packages/api/tests/endpoints/test-executions.js

```diff
@@ -1002,3 +1002,221 @@ test.serial('POST /executions with non-existing parentArn still creates a new ex
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(404);
+
+  const expectedErrorMessage = `Execution '${updatedExecution.arn}' not found`;
+  t.truthy(response.body.message.match(expectedErrorMessage));
+});
+
+test.serial('PUT /executions with non-existing asyncOperation throws error', async (t) => {
+  const execution = fakeExecutionFactoryV2();
+
+  const updatedExecution = fakeExecutionFactoryV2({
+    ...execution,
+    asyncOperationId: uuidv4(),
+  });
+
+  await request(app)
+    .post('/executions')
+    .send(execution)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  const response = await request(app)
+    .put(`/executions/${updatedExecution.arn}`)
+    .send(updatedExecution)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(400);
+
+  const expectedErrorMessage = `Record in async_operations .*${updatedExecution.asyncOperationId}.* does not exist`;
+  t.truthy(response.body.message.match(expectedErrorMessage));
+});
+
+test.serial('PUT /executions with non-existing collectionId throws error', async (t) => {
+  const execution = fakeExecutionFactoryV2();
+
+  const updatedExecution = fakeExecutionFactoryV2({
+    ...execution,
+    collectionId: constructCollectionId(randomId('name'), randomId('version')),
+  });
+
+  await request(app)
+    .post('/executions')
+    .send(execution)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  const response = await request(app)
+    .put(`/executions/${updatedExecution.arn}`)
+    .send(updatedExecution)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(400);
+
+  const expectedErrorMessage = 'Record in collections with identifiers .* does not exist';
+  t.truthy(response.body.message.match(expectedErrorMessage));
+});
+
+test.serial('PUT /executions with non-existing parentArn still updates the execution', async (t) => {
+  const execution = fakeExecutionFactoryV2();
+  const updatedExecution = fakeExecutionFactoryV2({
+    ...execution,
+    parentArn: randomId('parentArn'),
+  });
+
+  await request(app)
+    .post('/executions')
+    .send(execution)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  await request(app)
+    .put(`/executions/${updatedExecution.arn}`)
+    .send(updatedExecution)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  const fetchedDynamoRecord = await executionModel.get({
+    arn: updatedExecution.arn,
+  });
+
+  const fetchedPgRecord = await executionPgModel.get(
+    t.context.knex,
+    {
+      arn: updatedExecution.arn,
+    }
+  );
+
+  t.is(fetchedPgRecord.arn, fetchedDynamoRecord.arn);
+  t.falsy(fetchedPgRecord.parent_cumulus_id);
+});
```

Comment:
I'm sure it's just my ignorance I see the way api/endpoints/executions.js/update() doesn't check for the existence of the parentArn/parent_cumulus_id and how that results in the update not failing for non-existent parentArn, but shouldn't it fail? Again, probably just my ignorance.

Reply:
The current baseline ignores the parentArn is it doesn't exist.
https://github.com/nasa/cumulus/blob/master/packages/db/src/translate/executions.ts#L139

Reply:
Ah. Understood.